code/controllers/UserController.php

```diff
@@ -17,8 +17,9 @@ public function index()
     {
         if (!Member::currentUserID()) {
             return $this->redirect('/Security/login');
-        }
-        return $this->renderWith(array('UserController', 'Page'));
+        } else if (UserInvitation::create()->canCreate(Member::CurrentUser()) == false){
```

Comment:
Why don't we make this fully CMS configurable by letting this controller implement PermissionProvider, then providePermissions could look like this:
```php
    public function providePermissions()
    {
         return array(
            'ACCESS_USER_INVITATIONS' => array(
                'name' => _t('UserController.ACCESS_PERMISSIONS', 'Allow user invitations'),
                'category' => _t('UserController.CMS_ACCESS_CATEGORY', 'User Invitations')
            )
        );
    }
```

Then remove the security check from Index:
```php
        if (!Member::currentUserID() && UserInvitation::create()->canCreate()) {
            return $this->redirect('/Security/login');
        }
```

Then introduce an init with the security check to secure all actions on this controller
```php
public function init(){
parent::init();
if(!Permission::check('ACCESS_USER_INVITATIONS')){
   return Security::permissionFailure();
}
```

Thoughts?